ttljob: fix panic accessing TestingKnobs

otan opened the pull request:
Turns out it's not populated in production (but is in all the testing we
do!). Woops!

Resolves: https://github.com/cockroachdb/cockroach/issues/77840

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/78161)
<!-- Reviewable:end -->

rafiss reviewed (approved):
it could help to add a test that doesn't set the testing knobs so we have a regression test for this that gives quicker feedback.

otan commented:
Without testing knobs the test would take several minutes due to the nature of timing it.

otan commented:
bors r=rafiss

otan commented:
bors r-

Maybe I have an idea

otan committed:
ttljob: fix panic accessing TestingKnobs

Turns out it's not populated in production (but is in all the testing we
do!). Woops!

Release note: None

otan commented:
bors r=rafiss
